### PR TITLE
feat: add live code and formatting

### PR DESCRIPTION
**Pull request by Rei-x**
This pull request includes several changes to improve code formatting, add new dependencies, and update documentation. The most important changes include adding Prettier configuration, integrating new Astro plugins, and updating the `package.json` file with new scripts and dependencies.

### Code Formatting and Linting:
* [`.prettierrc`](diffhunk://#diff-663ade211b3a1552162de21c4031fcd16be99407aae5ceecbb491a2efc43d5d2R1-R8): Added Prettier configuration for consistent code formatting.
* [`.husky/pre-commit`](diffhunk://#diff-d2bc4bbf14eadc292d84e0ef5f7a93115c23557f533809fc80b896934291529dR1): Added a pre-commit hook to run `lint-staged`.
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R39-R48): Added `prettier`, `prettier-plugin-astro`, `husky`, and `lint-staged` to `devDependencies` and updated scripts to include `format` and `prepare`.

### Astro Configuration:
* [`astro.config.mjs`](diffhunk://#diff-e0f0c5adbe0b9ca5d0b57caf5cea33a8d88899fd02a43df1e9862b185f8a1e5fR6-R9): Integrated `astro-live-code` and `@astrojs/react` plugins, and updated the site configuration with new locales and favicon. [[1]](diffhunk://#diff-e0f0c5adbe0b9ca5d0b57caf5cea33a8d88899fd02a43df1e9862b185f8a1e5fR6-R9) [[2]](diffhunk://#diff-e0f0c5adbe0b9ca5d0b57caf5cea33a8d88899fd02a43df1e9862b185f8a1e5fR18-R31) [[3]](diffhunk://#diff-e0f0c5adbe0b9ca5d0b57caf5cea33a8d88899fd02a43df1e9862b185f8a1e5fR41) [[4]](diffhunk://#diff-e0f0c5adbe0b9ca5d0b57caf5cea33a8d88899fd02a43df1e9862b185f8a1e5fR64)

### Documentation Updates:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R12-R19): Added an example of a Markdown file with frontmatter for new documentation pages.
* [`src/content/docs/guides/Web Development/`](diffhunk://#diff-544812ca940edda007553f7c15cbcc28f19274886c007301ad7d71bab2aaaa5dR36-R47): Updated various `.md` and `.mdx` files to improve code examples, formatting, and added live code blocks using `astro-live-code`. [[1]](diffhunk://#diff-544812ca940edda007553f7c15cbcc28f19274886c007301ad7d71bab2aaaa5dR36-R47) [[2]](diffhunk://#diff-544812ca940edda007553f7c15cbcc28f19274886c007301ad7d71bab2aaaa5dR56-R67) [[3]](diffhunk://#diff-23d130ad4dcf4c25a57760ade5355097ef38d83eca79e8848392e05e30e9bf9dL6-R7) [[4]](diffhunk://#diff-23d130ad4dcf4c25a57760ade5355097ef38d83eca79e8848392e05e30e9bf9dL36-R37) [[5]](diffhunk://#diff-23d130ad4dcf4c25a57760ade5355097ef38d83eca79e8848392e05e30e9bf9dL60-R80) [[6]](diffhunk://#diff-23d130ad4dcf4c25a57760ade5355097ef38d83eca79e8848392e05e30e9bf9dL94-R102) [[7]](diffhunk://#diff-23d130ad4dcf4c25a57760ade5355097ef38d83eca79e8848392e05e30e9bf9dL109-R129) [[8]](diffhunk://#diff-23d130ad4dcf4c25a57760ade5355097ef38d83eca79e8848392e05e30e9bf9dR145-L139) [[9]](diffhunk://#diff-23d130ad4dcf4c25a57760ade5355097ef38d83eca79e8848392e05e30e9bf9dR223-R226) [[10]](diffhunk://#diff-8556c14065afe1973263453631ff88a688031f1c4709d118597d8475724e37b4R53-R57) [[11]](diffhunk://#diff-26b9219dbb3b0a63ffd4d24c2407a103c2b2f9e79e3d8cd90ba664b31cfc3350R5-R20) [[12]](diffhunk://#diff-26b9219dbb3b0a63ffd4d24c2407a103c2b2f9e79e3d8cd90ba664b31cfc3350L29-R148) [[13]](diffhunk://#diff-26b9219dbb3b0a63ffd4d24c2407a103c2b2f9e79e3d8cd90ba664b31cfc3350L154-R157) [[14]](diffhunk://#diff-26b9219dbb3b0a63ffd4d24c2407a103c2b2f9e79e3d8cd90ba664b31cfc3350L183-R213)

These changes enhance the project's code quality, streamline the development process, and improve the documentation for better readability and usability.

### CODE REVIEW

File: astro.config.mjs [hunk 1]
```diff
@@ -11,8 +15,20 @@ export default defineConfig({
...
+    }),
     starlight({
       defaultLocale: "pl",
+      locales: {
+        root: {
+          label: "Polski",
+          lang: "pl",
+        },
+      },
+      favicon: "favicon.ico",
```
Comment (qamarq) [inline astro.config.mjs:31]: będziemy robić multi-lang?

File: src/content/docs/guides/Web Development/dates.mdx [hunk 2]
```diff
@@ -2,18 +2,22 @@
...
-Nasz naczelny Żelownik Bartuś podrzucił libkę o nazwie **date-fns**. Popieram go całym serduszkiem i dzięki niej mam nadzięję sprawię, że to daty będą bały się Was a nie na odwrót.
+Uff, to jest ciężki temat. Każdy kiedyś zetknie się z datami (lub już się zetknął) i bez odpowiednich narzędzi potrafi to być czasem droga przez mękę. Te wszystkie strefy czasowe, wyświetlanie w różnych formatach, porównywanie itd.
+
+Nasz naczelny Żelownik Bartuś podrzucił libkę o nazwie [date-fns](https://date-fns.org/). Popieram go całym serduszkiem i dzięki niej mam nadzięję sprawię, że to daty będą bały się Was a nie na odwrót.
 
 :::tip
 Jeśli szukasz jakie literki wykorzystać do osiągnięcia zamierzonego efektu w formatowaniu, zapraszam do rozdziału niżej pt. "Dziwne literki do formatowania"
 :::
 
-## Basic usage
```
Comment (qamarq) [inline src/content/docs/guides/Web Development/dates.mdx:16]: good catch

**Review by qamarq — APPROVED** [review]
wiekszosc to jak widze formatowanie ale podoba mi sie meeega ten pomysl z live code i poprawki do dat. i love it, nic nie znalazlem podejrzanego, merguj 😘

File: astro.config.mjs (hunk 1 above)
Comment (Rei-x) [reply astro.config.mjs:31]: nope, to jest tylko po to, żeby wymusić język polski bo u mnie po angielsku było

**Pull request merged**